### PR TITLE
Fluentd DaemonSet needs a specific Label on Nodes

### PR DESCRIPTION
**Pull request by bbrowning**
The Fluentd DaemonSet will only run on nodes labeled with
`beta.kubernetes.io/fluentd-ds-ready="true"`. This label isn't set on
Nodes out of the box on some platforms, including minikube. It appears
to be set by default on GKE.

The Elasticsearch indices cannot be created until this label is set and Fluentd starts shipping logs.

There were a couple of broken anchor links I found in this same document while fixing this problem, so I've included that here as well.

### CODE REVIEW

**Comment by bbrowning** [comment]
For some reason I thought Prow would assign an approver here automatically. It appears not, so I'm assigning the one it suggested.
/assign @mchmarny

**Comment by bbrowning** [comment]
@RichieEscarez Please look at this change when you're look at the larger scope of logging docs. This config is still needed in at least some Kubernetes clusters, but I wonder how much detail we really want to get into with our Fluentd documentation in the Knative project itself.

**Comment by beckermarc** [comment]
I was also stumbling across this issue, while testing Knative on Kubernetes on Docker.
The part added to the documentation by thie PR solved the issue for me. Would appreciate, if this was part of the documentation.

Only thing I stumbled over is that in `kubectl get daemonset fluentd-ds -n monitoring` the namespace should meanwhile be `knative-monitoring` instead of just `monitoring`

File: serving/installing-logging-metrics-traces.md [hunk 1]
```diff
@@ -56,6 +56,29 @@ skip this step and continue to
 To visualize logs with Kibana, you need to set which Elasticsearch indices to explore. We will create two indices in Elasticsearch using `Logstash` for application logs and `Zipkin`
 for request traces.
 
+- Ensure your Kubernetes node(s) are labeled with
```
[the remark this answers is not in the document]
Comment (yanweiguo) [reply serving/installing-logging-metrics-traces.md]: Stackdriver also needs `beta.kubernetes.io/fluentd-ds-ready=true` label.

File: serving/installing-logging-metrics-traces.md [hunk 2]
```diff
@@ -12,44 +12,64 @@ these two are not supported.
...
-  node-exporter-cr6bh                   2/2       Running   0          2d
-  node-exporter-mf6k7                   2/2       Running   0          2d
-  node-exporter-rhzr7                   2/2       Running   0          2d
-  prometheus-system-0                   1/1       Running   0          2d
-  prometheus-system-1                   1/1       Running   0          2d
-  ```
-
-  CTRL+C to exit watch.
+   ```shell
+   kubectl get pods --namespace knative-monitoring --watch
```
[the remark this answers is not in the document]
Comment (RichieEscarez) [reply serving/installing-logging-metrics-traces.md]: Yes, thats right. Master in the docs repo represents the latest release, 0.2.x.

**Comment by RichieEscarez** [comment]
I know that the files have moved per https://github.com/knative/serving/pull/2169 but those changes need to be done in a separate PR

**Comment by yanweiguo** [comment]
FYI: There's another PR #613 to improve this doc.

File: serving/installing-logging-metrics-traces.md [hunk 3]
```diff
@@ -117,42 +137,63 @@ To configure and setup monitoring:
...
-        --filename third_party/config/monitoring/common \
-        --filename config/monitoring/200-common \
-        --filename config/monitoring/200-common/100-istio.yaml
-      ```
+   ```shell
+   kubectl apply --recursive --filename config/monitoring/100-common \
+     --filename config/monitoring/150-stackdriver \
+     --filename third_party/config/monitoring/common \
+     --filename config/monitoring/200-common \
+     --filename config/monitoring/200-common/100-istio.yaml
```
[the remark this answers is not in the document]
Comment (RichieEscarez) [reply serving/installing-logging-metrics-traces.md]: > Not for this PR, but it seems like these should be deployed from some sort of released URL, possibly one which has been aggregated.

I ask the same question about using the *.yaml files here too: #520 

> Also, all these files seem to have been moved and don't match these locations.

yeah, details of the move are in #2169 as i commented above but I didnt want to take on those changes in this PR

File: serving/installing-logging-metrics-traces.md [hunk 4]
```diff
@@ -117,42 +137,63 @@ To configure and setup monitoring:
...
+   ```
 
-   The installation is complete when logging & monitoring components are all
-   reported `Running` or `Completed`:
+     The installation is complete when logging & monitoring components are all
+     reported `Running` or `Completed`:
+
+    ```shell
+    kubectl get pods --namespace monitoring --watch
```
[the remark this answers is not in the document]
Comment (RichieEscarez) [reply serving/installing-logging-metrics-traces.md]: sorry, it looks like i lost and missed these during the manual merge ([my earlier commit](https://github.com/knative/docs/pull/332/commits/0ff2769681485b0df36679042cec1dba5af7106d))

**Pull request merged**